src/display/node_stream.js

```diff
@@ -288,8 +260,29 @@ class PDFNodeStreamRangeReader extends BaseRangeReader {
     this._request.end();
   }
 
+  read() {
+    return this._readCapability.promise.then(() => {
+      if (this._done) {
+        return Promise.resolve({ value: undefined, done: true, });
+      }
+      if (this._errored) {
+        return Promise.reject(this._reason);
+      }
+
+      let chunk = this._rangeRequest.read();
```

Comment:
This is duplication of code. The two implementations of `read` are identical.

What stops you from putting `read` in the base class?
If -for some reason- you cannot use `this._rangeRequest.read()` in the base class, then you can create a method in the subclass (e.g. `_read() { return this._rangeRequest.read(); }` and call `this._read();` from the base class instead.

And also, I thought that you had replaced "_rangeRequest" with "_readableStream", but I can not find that change (any more?).